### PR TITLE
Fix tf.nn.softmax

### LINKED ISSUE #8667: tf.nn.softmax -- the "name" argument is passed to the internal flattened tensor
Hi, if I use tf.nn.softmax with the `name` argument, I expect the output tensor to be given this name.
However, the output tensor is always a Reshape op named `Reshape_...`, and the passed name refers to an internal (flattened) tensor.

Steps to reproduce (TF v1.0.1):
```
In [1]: import tensorflow as tf
In [2]: a = tf.nn.softmax(tf.ones([2, 8, 20]), name="softmax_custom_name")
In [3]: print a
Tensor("Reshape_1:0", shape=(2, 8, 20), dtype=float32)
In [4]: print tf.get_default_graph().get_tensor_by_name("softmax_custom_name:0")
Tensor("softmax_custom_name:0", shape=(16, 20), dtype=float32)
```

Here is the relevant code from the master branch:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L1501
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/nn_ops.py#L1519

Is this the expected behaviour? In this case it would probably make sense to mention it in the documentation?

Otherwise I'd submit a fix.

### PR DESCRIPTION
**Pull request by ilya-eder**
Fixes #8667

### CODE REVIEW

File: tensorflow/python/ops/nn_ops.py [hunk 1]
```diff
@@ -1523,7 +1523,7 @@ def _swap_axis(logits, dim_index, last_index):
   # shape.
   output.set_shape(shape)
 
-  return output
+  return tf.identity(output, name=name)
```
Comment (skye) [inline tensorflow/python/ops/nn_ops.py]: Instead of adding an identity op, how about plumbing name through _swap_axis() to array_ops.tranpose() above?

**Comment by drpngx** [comment]
Jenkins, test this please.

@ilya-edrenkin any chance to take a look at the review comments?

File: tensorflow/python/ops/nn_ops.py (hunk 1 above)
Comment (ilya-edrenkin) [inline tensorflow/python/ops/nn_ops.py]: sure!

**Review by skye — APPROVED** [review]

**Comment by drpngx** [comment]
Jenkins, test this please.
